azurerm/internal/services/apimanagement/api_management_gateway_api_resource_test.go

```diff
@@ -0,0 +1,133 @@
+package apimanagement_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type ApiManagementGatewayAPIResource struct {
+}
+
+func TestAccApiManagementGatewayApi_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_gateway_api", "test")
+	r := ApiManagementGatewayAPIResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiManagementGatewayApi_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_gateway_api", "test")
+	r := ApiManagementGatewayAPIResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (ApiManagementGatewayAPIResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := azure.ParseAzureResourceID(state.ID)
```

Comment:
Can we use the `parse.NewGatewayID()` parser here and reference the resulting struct fields below?

Reply:
updated